homeassistant/components/coinbase/__init__.py

```diff
@@ -42,48 +47,66 @@
 )
 
 
-def setup(hass, config):
-    """Set up the Coinbase component.
-
-    Will automatically setup sensors to support
-    wallets discovered on the network.
-    """
-    api_key = config[DOMAIN][CONF_API_KEY]
-    api_secret = config[DOMAIN][CONF_API_SECRET]
-    account_currencies = config[DOMAIN].get(CONF_ACCOUNT_CURRENCIES)
-    exchange_currencies = config[DOMAIN][CONF_EXCHANGE_CURRENCIES]
-
-    hass.data[DATA_COINBASE] = coinbase_data = CoinbaseData(api_key, api_secret)
-
-    if not hasattr(coinbase_data, "accounts"):
-        return False
-    for account in coinbase_data.accounts.data:
-        if account_currencies is None or account.currency in account_currencies:
-            load_platform(hass, "sensor", DOMAIN, {"account": account}, config)
-    for currency in exchange_currencies:
-        if currency not in coinbase_data.exchange_rates.rates:
-            _LOGGER.warning("Currency %s not found", currency)
-            continue
-        native = coinbase_data.exchange_rates.currency
-        load_platform(
-            hass,
-            "sensor",
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Coinbase component."""
+    if DOMAIN not in config:
+        return True
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
             DOMAIN,
-            {"native_currency": native, "exchange_currency": currency},
-            config,
+            context={"source": SOURCE_IMPORT},
+            data=config[DOMAIN],
         )
+    )
 
     return True
 
 
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Coinbase from a config entry."""
+    client = await hass.async_add_executor_job(
+        Client,
+        entry.data[CONF_API_KEY],
+        entry.data[CONF_API_TOKEN],
+    )
+    hass.data[DOMAIN] = {}
```

Comment:
Since we can set up multiple entries, this one would clear previous setup entries now